### PR TITLE
7.0 fix timesheet fulfill crash

### PR DESCRIPTION
**Pull request by gurneyalex**
fix a crash with the timesheet fulfill wizard which was passing datetime objects instead of strings 

add a test showing the bug is fixed.

### CODE REVIEW

File: hr_timesheet_fulfill/wizard/timesheet_fulfill.py [hunk 1]
```diff
@@ -134,8 +133,7 @@ def fulfill_timesheet(self, cr, uid, ids, context=None):
                 # hh_mm is a tuple (hours, minutes)
                 date_end = " %d:%d:00" % (hh_mm[0], hh_mm[1])
                 att_end = {
-                    'name': datetime.strptime(datetime_current + date_end,
-                                              '%Y-%m-%d %H:%M:%S'),
+                    'name': datetime_current + date_end,
```
Comment (pedrobaeza) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: There is one missing space between both strings.

Comment (gurneyalex) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: @pedrobaeza I must be tired, I really don't see what you mean there. Could you clarify?

Comment (pedrobaeza) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: OK, don't worry, I try to explain:

datetime_current = "2014-01-01"
date_end = "22:46:00"

datetime_current + date_end = "2014-01-0122:46:00", not "2014-01-01 22:46:00"

Regards.

Comment (gurneyalex) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: I'm quite certain there is a space after the opening quote on line 134

**Comment by gurneyalex** [comment]
It looks like the added test is triggering an error in `hr_attendance_analysis`. I'll look into this and try to provide a fix.

File: hr_timesheet_fulfill/wizard/timesheet_fulfill.py (hunk 1 above)
Comment (pedrobaeza) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: Sorry, I didn't see it. I only see the concatenation of the two strings.

**Comment by gurneyalex** [comment]
the error was there before: it is a bug in the test which only fires if the tests are run after the 11th of the month... See #10 for the fix.

File: hr_timesheet_fulfill/test/timesheet_fulfill.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    description: vacations
+    nb_hours: 8.0
+    analytic_account_id: account_vacations
+-
+  run wizard
+-
+  !python {model: hr.timesheet.fulfill }: |
+    context['active_id'] = ref('test_timesheet')
+    ret = self.fulfill_timesheet(cr, uid, [ref('wizard_fulfill')], context=context)
+    assert ret == {'type': 'ir.actions.act_window_close'}
```
Comment (eLBati) [inline hr_timesheet_fulfill/test/timesheet_fulfill.yml:34]: Is it possible to make the test check the created data?

Comment (gurneyalex) [inline hr_timesheet_fulfill/test/timesheet_fulfill.yml:34]: @elBati: you're right, I was focusing on the crash when writing the test. I'll update this.

**Comment by gurneyalex** [comment]
Travis is read because of 80char lines in other modules. See #12 for this.

File: hr_timesheet_fulfill/wizard/timesheet_fulfill.py [hunk 3]
```diff
@@ -123,23 +127,23 @@ def fulfill_timesheet(self, cr, uid, ids, context=None):
...
                     'action': 'sign_out',
                     'employee_id': employee_id,
-                    'sheet_id': timesheet.id,
                 }
-                attendance_obj.create(cr, uid, att_start, context)
-                attendance_obj.create(cr, uid, att_end, context)
+                attendance_obj.create(cr, uid, att_start,
+                                      ontext=attendance_context)
```
Comment (eLBati) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py]: TypeError: create() got an unexpected keyword argument 'ontext'

**Comment by eLBati** [comment]
@gurneyalex please rebase. Thanks

**Comment by gurneyalex** [comment]
Rebased. Travis is red because of the new pylint config (handled in #16).

**Comment by guewen** [comment]
:+1:

File: hr_timesheet_fulfill/wizard/timesheet_fulfill.py (hunk 1 above)
Comment (lepistone) [inline hr_timesheet_fulfill/wizard/timesheet_fulfill.py:140]: :weary: I have a bit of a headache, you're saying `date_end` is actually a time, and `datetime_current` is actually a date? (string)

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by lepistone** [comment]
I merged #16 and restarted travis.

**Comment by lepistone** [comment]
Ok, you didn't introduce the weird variable names, you're fixing other stuff. :+1: , and I'm merging, thanks!

**Pull request merged**